Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -314,11 +319,69 @@ void CookieStore::remove(CookieStoreDeleteOptions&& options, Ref<DeferredPromise
     set(WTFMove(initOptions), WTFMove(promise));
 }
 
+void CookieStore::cookiesAdded(const String& host, const Vector<Cookie>& cookies)
+{
+    ASSERT_UNUSED(host, host == m_host);
+    if (!m_hasChangeEventListener)
+        return;
+
+    auto* context = scriptExecutionContext();
+    if (!context)
+        return;
+
+    CookieChangeEventInit eventInit;
+    auto currentTime = WallTime::now().secondsSinceEpoch().milliseconds();
+    for (auto cookie : cookies) {
+        if (cookie.expires && *cookie.expires < currentTime) {
+            cookie.value = nullString();
+            eventInit.deleted.append(CookieListItem { cookie });
+        } else
+            eventInit.changed.append(CookieListItem { cookie });
+    }
+
+    queueTaskToDispatchEvent(*this, TaskSource::DOMManipulation, CookieChangeEvent::create(eventNames().changeEvent, WTFMove(eventInit), CookieChangeEvent::IsTrusted::Yes));
+}
+
+void CookieStore::cookiesDeleted(const String& host, const Vector<Cookie>& cookies)
+{
+    ASSERT_UNUSED(host, host == m_host);
+    if (!m_hasChangeEventListener)
```

Comment:
Should this be an assertion? Why would we be getting notifications if we don't have a change event listener?